api/client.go

```diff
@@ -453,6 +508,11 @@ func (c *Client) NewRequest(method, requestPath string) *Request {
 // a Vault server not configured with this client. This is an advanced operation
 // that generally won't need to be called externally.
 func (c *Client) RawRequest(r *Request) (*Response, error) {
+	c.modifyLock.RLock()
+	c.config.modifyLock.RLock()
+	defer c.config.modifyLock.RUnlock()
+	c.modifyLock.RUnlock()
```

Comment:
I think it wouldn't hurt if we defer here.

Reply:
Doing so would mean holding the lock for the lifetime of the client request, including redirects and timeouts, even though we no longer need to access anything it's protecting.

Reply:
You’re right, holding the client lock would be unnecessary.